wasm/names_test.go

```diff
@@ -0,0 +1,129 @@
+package wasm
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/require"
+)
+
+func TestEncodeNameSection(t *testing.T) {
+	m := &Module{
+		Name:        "simple",
+		TypeSection: []*FunctionType{{}},
+		ImportSection: []*ImportSegment{
+			{
+				Module: "",
+				Name:   "Hello",
+				Desc: &ImportDesc{
+					Kind:          ImportKindFunction,
+					FuncTypeIndex: 0,
+					FuncName:      "hello",
+				},
+			},
+		},
+	}
+
+	// TIP: the below is the binary suffix of `wat2wasm --debug-names --debug-parser -v simple.wat` where simple.wat
+	// contains the same text as the above inlined text format. Ex.
+	//	(module $simple
+	//		(import "" "Hello" (func $hello))
+	//		(start $hello)
+	//	)
+	require.Equal(t, []byte{
+		SectionIDCustom,
+		0x1d, // 24 bytes after this point
+		0x04, // the custom second name "name" is 4 bytes long
+		'n', 'a', 'm', 'e',
+		0x00, // module subsection ID zero
+		0x07, // 7 bytes to follow
+		0x06, // the module name simple is 6 bytes long
```

Comment:
due to these changes, to add the section prefix, and the changes to append an empty local subsection, the git diff considers this a new file